src/addresses.ts

```diff
@@ -0,0 +1,48 @@
+import BN from 'bn.js';
+
+const { addressIdToAddress, addressToAddressId } = require("substrate-ss58");
+const L1ADDR_BITS = 160;
+
+export function encodeL1address(addressHex: string, chex: string) {
+  let c = new BN(chex + "0000000000000000000000000000000000000000", "hex");
+  let a = new BN(addressHex, 16);
+  return c.add(a);
+}
+
+/* chain_id:dec * address:hex
+ */
+export function decodeL1address(l1address: string) {
+  let uid = new BN(l1address);
```

Comment:
Sometimes users will provide an invalid l1address. Maybe put a TODO here so later we can add a valid check here.